### PR TITLE
perf(topk): optimize persistent radix top-k for long sequences

### PR DESCRIPTION
**Pull request by tuanzhangCS**
## Summary

- Replace the scheduler-heavy persistent radix top-k path with per-row register and streaming kernels plus length-specialized cluster-8 and cluster-16 paths for long rows.
- Use the actual live length for each row, including causal verify lengths, so CUDA Graph padding does not force scans over inactive columns.
- Route the Qwen4-Exp materialized-score QSA path through the persistent CUDA selector with caller-owned workspace while retaining the portable Triton fallback.
- Keep long-row launches CUDA Graph safe with programmatic dependent launch, and fall back to the original single-CTA kernel when cluster launches are unsupported or rejected.

## Performance

Measured on one NVIDIA B200 (`sm_100a`, CUDA 13.0). The baseline is the PR merge-base (`9104ebb6`); the PR result is `bdd1d33f`. Inputs use FP32 logits and near-full ragged row lengths. Each result is the median of 7 alternating rounds, with 300 launches per round after 50 warmup launches, timed with CUDA events. Outputs from both implementations were checked against `torch.topk` outside the timed region.

| Rows | Width | Top-k | Baseline (us) | PR (us) | Speedup |
| ---: | ---: | ---: | ---: | ---: | ---: |
| 1 | 65,536 | 512 | 22.450 | 8.908 | 2.52x |
| 8 | 65,536 | 512 | 26.669 | 9.103 | 2.93x |
| 1 | 131,072 | 2,048 | 24.601 | 9.918 | 2.48x |
| 8 | 131,072 | 2,048 | 28.011 | 10.126 | 2.77x |
| 1 | 262,144 | 2,048 | 26.646 | 11.679 | 2.28x |
| 8 | 262,144 | 2,048 | 32.788 | 15.707 | 2.09x |

Across these long-context cases, the persistent top-k path is **2.09x to 2.93x faster** than the merge-base implementation.

## Test Plan

- `pre-commit run --all-files`
- Rebuild the CUDA extension for `sm_100a`.
- Targeted runtime workspace test: 1 passed.
- Targeted persistent top-k and QSA CUDA tests: 5 passed, with 10 additional parameterized subtests passed.
- Validate register, streaming, cluster-8, cluster-16, CUDA Graph replay, ragged-length, QSA dispatch, and non-cluster fallback paths.

### CODE REVIEW

File: start.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+ts serve \
```
[the remark this answers is not in the document]
Comment (tuanzhangCS) [reply start.sh]: ok
